src/plugins/intel_gna/src/transformations/aszp_decomposition.cpp

```diff
@@ -0,0 +1,231 @@
+    std::tie(N,H,W,C) = GetInputDimensions(conv_input_shape);
+
+	if (Hpad != 0) {
+		auto new_reshape   = CreateReshape(input, 2, Shape{H,W*C});
+		auto new_transpose = CreateTranspose(new_reshape->output(0));
+
+		auto padding_const = CreateZeroConst(Shape{W*C,Hpad});
+		auto new_concat    = ConcatZeros(Hbegin,Hend,padding_const,new_transpose);
+		
+		auto new_untranspose = CreateTranspose(new_concat->output(0));	// Shape {W*C,H+Hpad}
+
+		if (0 == Wpad) return CreateReshape(new_untranspose->output(0), 4, Shape{N, H+Hpad, W ,C})->output(0);
+		else return (new_untranspose->output(0));
+	}else return input;
+
+
+}
+
+
+
+Output<Node> DecomposeWidth(Output<Node> input, ov::CoordinateDiff pads_begin, ov::CoordinateDiff pads_end, ov::Shape conv_input_shape) {
+	uint32_t Hbegin, Hend, Wbegin, Wend, Hpad, Wpad;
+	uint64_t N, H, W, C;
+    std::tie(Hbegin, Hend, Wbegin, Wend, Hpad, Wpad) = ExtractPadding(pads_begin, pads_end); //todo cleanup. maybe get only what we need?
+    std::tie(N,H,W,C) = GetInputDimensions(conv_input_shape);
+
+	if (Wpad != 0) {
+		auto new_reshape   = CreateReshape(input, 2, Shape{(H+Hpad)*W, C});
+		auto new_transpose = CreateTranspose(new_reshape->output(0));
+		auto new_reshape2  = CreateReshape(new_transpose->output(0),2, Shape{C*(H+Hpad), W});
+
+		auto padding_const = CreateZeroConst( Shape{C*(H+Hpad),Wpad} );
+		auto new_concat    = ConcatZeros(Wbegin,Wend,padding_const,new_reshape2);
+
+		auto new_unshape2    = CreateReshape(new_concat->output(0), 2, Shape{C, (H + Hpad) * (W + Wpad)});
+		auto new_untranspose = CreateTranspose(new_unshape2->output(0));
+		auto new_unshape     = CreateReshape(new_untranspose->output(0), 4, {N, H+Hpad, W+Wpad ,C});
+
+		return new_unshape->output(0);
+	}else return input;
+}
+
+
+
+void TrimmPadding(ov::CoordinateDiff& pads_begin, ov::CoordinateDiff& pads_end) {
+	if ( pads_begin[0] > pads_end[0]) 
+		 pads_begin[0] = pads_end[0];
+	else pads_end[0] = pads_begin[0];
+	if ( pads_begin[1] > pads_end[1]) 
+		 pads_begin[1] = pads_end[1];
+	else pads_end[1] = pads_begin[1];
+}
+
+
+std::shared_ptr<ov::Node> CreateConvolution (
+	std::shared_ptr<ov::Node> conv, 
+	const Output<Node>& input, 
+	ov::CoordinateDiff pads_begin, 
+	ov::CoordinateDiff pads_end){
+
+	TrimmPadding(pads_begin, pads_end);
+
+	auto cnn_node = std::dynamic_pointer_cast<ngraph::opset8::Convolution>(conv);
+	if (nullptr != cnn_node){
+		return std::make_shared<opset8::Convolution>(
+			input, 
+			cnn_node->input_value(1), 
+			cnn_node->get_strides(), 
+			pads_begin, pads_end, 
+			cnn_node->get_dilations(), 
+			cnn_node->get_auto_pad());
+	}
+
+	return nullptr;
+}
+
+
+// TODO ask about the input type. Convolution or Node. How to generalize for groupconv, dilated convolution
+static bool Decompose(std::shared_ptr<ngraph::opset8::Convolution> conv) {
+	auto pads_begin = conv->get_pads_begin();
+    auto pads_end   = conv->get_pads_end();
+	if (pads_begin[0] == pads_end[0] && pads_begin[1] == pads_end[1]) return false;
```

Comment:
shouldn't we check if pads_begin and pads_end lengths is at lease 2 to avoid access of not existing index

Reply:
Done